pkg/kubelet/dockershim/helpers.go

```diff
@@ -393,3 +396,48 @@ type dockerOpt struct {
 func (d dockerOpt) GetKV() (string, string) {
 	return d.key, d.value
 }
+
+// writeLimiter limits the total output written across one or more streams.
+type writeLimiter struct {
+	delegate io.Writer
+	limit    *int64
+}
+
+func (w writeLimiter) Write(p []byte) (int, error) {
+	if len(p) == 0 {
+		return 0, nil
+	}
+	limit := atomic.LoadInt64(w.limit)
+	if limit <= 0 {
+		return 0, errMaximumWrite
+	}
+	var truncated bool
+	if limit < int64(len(p)) {
+		p = p[0:limit]
+		truncated = true
+	}
+	n, err := w.delegate.Write(p)
+	if n > 0 {
+		newLimit := limit - int64(n)
+		for !atomic.CompareAndSwapInt64(w.limit, limit, newLimit) {
```

Comment:
It's arguable that we don't actually need a `for` loop to accomplish what this does. I originally wrote this because I thought there'd be more "work" to do in the loop iterations. But the result is over-complicated and only serves to decrease readability. e.g. it would be more straight forward to write `atomic.AddInt64(w.limit, -1 * n)`